### PR TITLE
Keeper V2

### PR DESCRIPTION
**Pull request by samirsalem**
List of changes:
- lighter build of apache airflow image (removed some uused stuff)
- secured install of the keeper
    - customize password of airflow user, pg
    - autogeneration of the fernet key for metadata encryption
    - nginx entry point with self-signed ssl certificate to secure GUI
    - introduced fail2ban to avoid ssh bruteforce
- automate deploy on custom platform via make script
- automate deploy via terraform on aws for CI
- automate sql tables creation
- automate import of variables and connections
- update to the latest python version and pg
- introduced loki for centralized logging  (WIP, not functional at the moment)
- introduced statsd exporter for monitoring (WIP, not functional at the moment)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -36,7 +36,7 @@
 # Use the same builder frontend version for everyone
 # syntax=docker/dockerfile:1.3
 ARG AIRFLOW_VERSION="2.2.3"
-ARG AIRFLOW_EXTRAS="amazon,async,celery,cncf.kubernetes,dask,docker,elasticsearch,ftp,google,google_auth,grpc,hashicorp,http,ldap,microsoft.azure,mysql,odbc,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv"
+ARG AIRFLOW_EXTRAS="async,celery,cncf.kubernetes,dask,docker,grpc,hashicorp,http,mysql,odbc,postgres,redis,sendgrid,sftp,ssh,statsd,virtualenv"
```
Comment (mikeghen) [inline Dockerfile]: still don't need MySQL

File: README.md [hunk 2]
```diff
@@ -1,79 +1,98 @@
...
+key5 = ""                               # address5 key
+key6 = ""                               # address6 key
+key7 = ""                               # address7 key
+address1 = ""                           # address1
+address2 = ""                           # address2
+address3 = ""                           # address3
+address4 = ""                           # address4
+address5 = ""                           # address5
+address6 = ""                           # address6
+address7 = ""                           # address7
```
Comment (mikeghen) [inline README.md]: Let's map these to the names of what they are in the GUI

File: templates/.vars.tmpl [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export HARVESTER_ADDRESS='${address5}'
+export REPORTER_ADDRESS='${address6}'
+export REX_BANK_KEEPER_ADDRESS='${address7}'
+export SWAPPER_ADDRESS_KEY='${key1}'
+export CLOSER_ADDRESS_KEY='${key2}'
+export DISTRIBUTOR_ADDRESS_KEY='${key3}'
+export DISTRIBUTOR_V2_ADDRESS_KEY='${key4}'
+export HARVESTER_ADDRESS_KEY='${key5}'
+export REPORTER_ADDRESS_KEY='${key6}'
+export REX_BANK_KEEPER_ADDRESS_KEY='${key7}'
```
Comment (mikeghen) [inline templates/.vars.tmpl]: Let's use something other than addressX and keyX

File: templates/docker-compose.tmpl.yml [hunk 4]
```diff
@@ -1,57 +1,27 @@
...
+    AIRFLOW__METRICS__STATSD_HOST: localhost
+    AIRFLOW__METRICS__STATSD_ON: 'true'
+    AIRFLOW__METRICS__STATSD_PORT: 9125
+    AIRFLOW__METRICS__STATSD_PREFIX: airflow
+    AIRFLOW__METRICS__STATSD_ALLOW_LIST: scheduler,executor,dagrun,dag_processing,dag,pool,ti
     AIRFLOW__CORE__EXECUTOR: CeleryExecutor
-    AIRFLOW__CORE__SQL_ALCHEMY_CONN: postgresql+psycopg2://airflow:airflow@postgres/airflow
-    AIRFLOW__CELERY__RESULT_BACKEND: db+postgresql://airflow:airflow@postgres/airflow
+    AIRFLOW__CORE__SQL_ALCHEMY_CONN: postgresql+psycopg2://airflow:postgres_password@postgres/airflow
+    AIRFLOW__CELERY__RESULT_BACKEND: db+postgresql://airflow:postgres_password@postgres/airflow
```
Comment (mikeghen) [inline templates/docker-compose.tmpl.yml:19]: Do these passwords get overriden?

**Review by mikeghen — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (samirsalem) [reply Dockerfile]: got it, wasn't sure if those was use

File: templates/docker-compose.tmpl.yml (hunk 4 above)
Comment (samirsalem) [reply templates/docker-compose.tmpl.yml:19]: yes those passwords gets overriden

**Comment by samirsalem** [comment]
requested changes have been pushed, my keeper is now running this version, testing new version

File: Dockerfile (hunk 1 above)
Comment (samirsalem) [reply Dockerfile]: done and removed some additional extras

File: README.md (hunk 2 above)
Comment (samirsalem) [reply README.md]: done

File: templates/.vars.tmpl (hunk 3 above)
Comment (samirsalem) [reply templates/.vars.tmpl]: done

File: templates/docker-compose.tmpl.yml (hunk 4 above)
Comment (samirsalem) [reply templates/docker-compose.tmpl.yml:19]: that's why this docker-compose file is a template, while deploying this is substituted with the user's choice

**Review by samirsalem — COMMENTED** [review]
my new keeper is running fine, both methods have been tested

**Comment by samirsalem** [comment]
@mikeghen this can me merged, you can test it to make sure that everything is fine by rebuilding your keeper

**Review by shreyaspapi — APPROVED** [review]

**Pull request merged**